tests/conftest.py

```diff
@@ -5370,5 +5365,36 @@ def finalizer():
         for instance in fioscale_instances:
             instance.cleanup()
 
+
+@pytest.fixture(scope="function")
+def switch_to_provider_for_test_factory(request):
+    """
+    Switch to provider cluster in case we use multi cluster and Managed service. If we don't use
+    multi cluster and Managed Services, the test will run as usual.
+
+    """
+    orig_index = None
+
+    def factory():
+        nonlocal orig_index
+
+        if (
+            config.multicluster
+            and config.ENV_DATA.get("platform", "").lower()
+            in constants.MANAGED_SERVICE_PLATFORMS
+        ):
+            log.info("Switching to the provider cluster context")
+            config.switch_to_provider()
+            orig_index = config.cur_index
```

Comment:
Sorry, maybe I misunderstand the logic, but shouldn't you save the original index before switching to provider?